### PR TITLE
MM-26836: Remove some app fields which cause race conditions

### PR DESCRIPTION
**Pull request by agnivade**
`(a *App) InitServer` was being called from multiple goroutines
and were modifying the fields of the App struct concurrently.
However, the fields were just reflecting the Server fields,
and we already have public getters for those fields.

Therefore, we remove those fields and just change the code to return
the server field directly.

### CODE REVIEW

**Comment by agnivade** [comment]
Well, this was easier than I thought. It's good that the entire codebase uses those getters.

**Comment by agnivade** [comment]
@ogi-m - Some general smoke testing would be sufficient here. Thanks.

File: app/options.go [hunk 1]
```diff
@@ -100,7 +100,6 @@ func ServerConnector(s *Server) AppOption {
 		a.log = s.Log
 		a.notificationsLog = s.NotificationsLog
 
-		a.accountMigration = s.AccountMigration
 		a.cluster = s.Cluster
```
Comment (jespino) [inline app/options.go:103]: Not for this PR, but would be great to get rid of all this when is possible.

Comment (agnivade) [reply app/options.go:103]: https://mattermost.atlassian.net/browse/MM-27169

**Comment by agnivade** [comment]
@ogi-m - Wondering if you had a chance to look at this? :)

**Pull request merged**